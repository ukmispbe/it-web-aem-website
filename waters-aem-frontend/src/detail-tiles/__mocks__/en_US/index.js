const DefaultProps = {
    name: 'detailTile',
    title: 'Detail Tile',
    type: 'personal',
    canCreate: true,
    addTitle: 'Add New Detail Tile',
    noAddressMessage: 'There are no tiles, add one now.',
    fetchEndPoint: 'www.waters.com',
    formMessage: {
        text: 'This is the form message'
    },
    icons: {
        add: '/content/dam/waters/en/brand-assets/icons/add.svg',
        edit: '/content/dam/waters/en/brand-assets/icons/edit-2.svg',
        refresh: '/content/dam/waters/en/brand-assets/icons/refresh.svg'
    },
    form: {
        buttonText: 'Save Changes',
        cancelText: 'Cancel',
        submitEndpoint: 'www.waters.com',
        icons: {
            invalidIcon:
                '/content/dam/waters/en/brand-assets/icons/attention.svg',
            validIcon:
                '/content/dam/waters/en/brand-assets/icons/checkmark.svg',
            lockIcon: '/content/dam/waters/en/brand-assets/icons/lock.svg'
        },
        fields: [
            {
                name: 'field1',
                label: 'Field One (Required)',
                type: 'text',
                validation: {
                    required: true,
                    requiredMsg: 'This field is required.',
                    validateFnName: 'noWhitespaceOnly',
                    validationMsg: 'Please enter some text.'
                }
            },
            {
                name: 'field2',
                label: 'Field Two (Optional)',
                type: 'text',
                validation: {
                    required: false,
                    validateFnName: 'noWhitespaceOnly',
                    validationMsg: 'Please enter some text.'
                }
            },
            {
                name: 'field3',
                label: 'Field Three (Disabled)',
                type: 'text',
                disabled: true,
                validation: {
                    required: true,
                    requiredMsg: 'This field is required.',
                    validateFnName: 'noWhitespaceOnly',
                    validationMsg: 'Please enter some text.'
                }
            }
        ]
    },
    tileName: 'tileDetailsTile',
    columns: [
        {
            title: 'Column 1 title',
            rows: [
                {
                    text: 'one line of text',
                    class: 'one-line'
                }
            ]
        },
        {
            title: '',
            rows: [
                {
                    text: 'There is',
                    class: 'there-is'
                },
                {
                    text: 'multiple lines of',
                    class: 'multiple-lines-of'
                },
                {
                    text: 'text',
                    class: 'text'
                }
            ]
        }
    ],
    defaultValues: {
        field1: 'Required Text',
        field3: 'Disabled Text'
    },
    notification: {
        title: 'Notification Title',
        description: 'Description for notification',
        icon: '/content/dam/waters/en/brand-assets/icons/refresh.svg'
    }
};

export default DefaultProps;
