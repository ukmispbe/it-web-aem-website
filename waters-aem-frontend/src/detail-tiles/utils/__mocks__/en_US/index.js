const defaultData = {
    firstName: "John",
    lastName: "Doe",
    localeCountry: "us",
    company: "Waters Corporation",
    phone: "1234567890",
    email: "john_doe@example.com",
    icon: "/content/dam/waters/en/brand-assets/icons/refresh.svg",
    userAddress: [
        {
            id: 0,
            preferred: true,
            company: "Waters Corp",
            street: "123 Main Street",
            city: "Cityville",
            stateRegion: "Stateton",
            zip: "12345",
            country: "United States",
            countryCode: "us",
            pending: false,
            addressType: "shippingAddress"
        },
        {
            id: 1,
            preferred: false,
            company: "Waters Corpss",
            street: "456 Main Street",
            city: "Cityville",
            stateRegion: "Stateton",
            zip: "67890",
            country: "United States",
            countryCode: "us",
            pending: true,
            addressType: "shippingAddress"
        },
        {
            id: 2,
            preferred: false,
            company: "Waters Corps",
            street: "789 Main Street",
            city: "Cityville",
            stateRegion: "Statetone",
            zip: "12345",
            country: "United States",
            countryCode: "us",
            pending: false,
            addressType: "shippingAddress"
        },
        {
            id: 3,
            preferred: false,
            company: "Waters Corps",
            street: "101 Main Street",
            city: "Cityville",
            stateRegion: "Stateton",
            zip: "12345",
            country: "United States",
            countryCode: "us",
            pending: false,
            addressType: "mailingAddress"
        },
        {
            id: 4,
            preferred: true,
            company: "Waters Corp.",
            street: "112 Main Street",
            city: "Cityville",
            stateRegion: "Stateton",
            zip: "10111",
            country: null,
            countryCode: "in",
            pending: true,
            addressType: "billingAddress"
        },
        {
            id: 5,
            preferred: false,
            company: "Waters Corp",
            street: "131 Main Street",
            city: "Cityville",
            stateRegion: "Stateton",
            zip: "21314",
            country: "United States",
            countryCode: "us",
            pending: true,
            addressType: "billingAddress"
        },
    ]
};

export default defaultData;