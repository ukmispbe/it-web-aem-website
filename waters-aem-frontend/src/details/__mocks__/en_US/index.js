const props = {
        "config": {
               "fetchDetailsEndPoint": "https://stgservices.waters.com/api/waters/order/v1/details/",
               "fetchItemsEndPoint": "https://stgservices.waters.com/api/waters/order/v1/details/",
               "historyTitle": "Order History",
               "detailsTitle": "Order Details",
               "resultNotFoundErrorTitle": "Sorry, order not found.",
               "serviceErrorNotificationTitle": "Sorry, something went wrong.",
               "serviceErrorNotificationText": "Please try again.",
               "serviceErrorNotificationIcon": "/content/dam/waters/en/brand-assets/icons/attention.svg",
               "orderNumber": "Order Number",
               "shipTo": "Ship to",
               "billTo": "Bill to",
               "summaryTitle": "Order Summary",
               "paymentMethod": "Payment Method",
               "shipmentMethod": "Shipment Method",
               "subTotal": "Subtotal",
               "items": "items",
               "shipping": "Shipping",
               "tax": "Tax",
               "orderTotal": "Order Total",
               "paymentType": {
                 "purchaseOrder": {
                   "label": "PO",
                   "icon": "/content/dam/waters/en/brand-assets/icons/document.svg"
                 },
                 "creditCard": {
                   "label": "Credit Card",
                   "icon": "/content/dam/waters/en/brand-assets/icons/creditcard.svg"
                 }
               }
        }
    }

export default props;