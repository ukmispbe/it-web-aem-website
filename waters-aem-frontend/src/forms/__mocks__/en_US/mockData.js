export const emilConfig = {
    type: "email",
    name: "email",
    label: "Email Address",
    description: "This will be your username.",
    validation: {
        required: true,
        requiredMsg: "Please enter a valid email address.",
        validateFnName: "email",
        validationMsg: "Please enter a valid email address.",
        alreadyRegisteredMsg: "There is an existing account with that email address.",
        signInMsg: "Sign In",
        signInURL: ""
    }
}
 
export const registrationConfig = {
    submitEndpoint: "https://test-www.waters.com:8443/api/waters/user/v1/registration",
    buttonText: "Create Account",
    icons: {
        checkmarkIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        validIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        invalidIcon: "/content/dam/waters/en/brand-assets/icons/attention.svg",
        eyeIcon: "/content/dam/waters/en/brand-assets/icons/eye.svg",
        eyeOffIcon: "/content/dam/waters/en/brand-assets/icons/eye-off.svg",
        signInIcon: "/content/dam/waters/en/brand-assets/icons/user.svg"
    },
    fields: [
        {
            type: "email",
            name: "email",
            label: "Email Address",
            description: "This will be your username.",
            emailValidationEndpoint: "https://test-www.waters.com:8443/api/waters/user/v1/validate/{email}",
            validation: {
                required: true,
                requiredMsg: "Please enter a valid email address.",
                validateFnName: "email",
                validationMsg: "Please enter a valid email address.",
                alreadyRegisteredMsg: "There is an existing account with that email address.",
                signInMsg: "Sign In",
                signInURL: ""
            }
        },
        {
            type: "text",
            name: "firstName",
            label: "First Name",
            validation: {
                required: true,
                validateFnName: "noWhitespaceOnly",
                validationMsg: "Please enter a valid first name.",
                requiredMsg: "Please enter a first name."
            }
        },
        {
            type: "text",
            name: "lastName",
            label: "Last Name",
            validation: {
                required: true,
                validateFnName: "noWhitespaceOnly",
                validationMsg: "Please enter a valid last name.",
                requiredMsg: "Please enter a last name."
            }
        },
        {
            type: "text",
            name: "company",
            label: "Company",
            validation: {
                validateFnName: "noWhitespaceOnly",
                validationMsg: "Please enter a valid company name.",
                required: true,
                requiredMsg: "Please enter a company name."
            }
        },
        {
            type: "dropdown",
            name: "country",
            label: "Country",
            placeholder: "Select...",
            dropdownIndicator: "/content/dam/waters/en/brand-assets/icons/down.svg",
            options: [{countryCode:"gb",displayName:"United Kingdom"},{countryCode:"us",displayName:"United States"}],
            validation: {
                required: true,
                requiredMsg: "Please select your country."
            }
        },
        {
            type: "password",
            name: "password",
            label: "Create Password",
            hasMatch: true,
            matchLabel: "Confirm Password",
            validation: {
                required: true,
                validateFnName: "password",
                validationMsg: "Please enter a valid password.",
                requiredMsg: "Please enter a password.",
                nonMatchingMsg: "Passwords must match. Please try again.",
                requirementsLabel: "Your password must include",
                requirements: [
                    {name: "shortPassword", msg: "at least 8 characters"},
                    {name: "noUppercase", msg: "at least 1 uppercase letter"},
                    {name: "noLowercase", msg: "at least 1 lowercase letter"},
                    {name: "noDigits", msg: "at least 1 number"},
                    {name: "noSpecial", msg: "at least 1 symbol (for example, !, $, #, %)"}
                ]
            }
        },
        {
            type: "checkbox",
            name: "communications",
            label: "I agree to receive marketing, product, training, support, and related communications via mail, email, or other electronic means."
        },
        {
            type: "captcha",
            name: "captcha",
            siteKey: "6Ld5WMIUAAAAACZQvEc7I75aEg5AC8YUUO0W7zRG",
            validation: {
                required: true
            }
        },
        {
            type: "textwithlinks",
            name: "privacy",
            config: [
                {
                    type: "text",
                    text: "By clicking the CREATE ACCOUNT button, you are agreeing to our"
                },
                {
                    type: "link",
                    label: "Terms of Use",
                    url: "",
                    blank: true
                },
                {
                    type: "text",
                    text: "and understand that the information you have provided on this website will be used and processed by Waters in accordance with the"
                },
                {
                    type: "link",
                    label: "Waters Privacy Notice.",
                    url: "",
                    blank: true
                }
            ]
        },
    ]
}

export const changePasswordConfig = {
    submitEndpoint: "https://test-www.waters.com:8443/api/waters/user/v1/registration",
    buttonText: "Change Password",
    cancelText: "Cancel",
    icons: {
        checkmarkIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        validIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        invalidIcon: "/content/dam/waters/en/brand-assets/icons/attention.svg",
        eyeIcon: "/content/dam/waters/en/brand-assets/icons/eye.svg",
        eyeOffIcon: "/content/dam/waters/en/brand-assets/icons/eye-off.svg",
        signInIcon: "/content/dam/waters/en/brand-assets/icons/user.svg"
            },
    fields: [
                {
            type: "password",
            name: "currentPassword",
            label: "Enter Current Password",
            hasMatch: false,
            validation: {
                required: true,
                requiredMsg: "Please enter current password."
                    }
                },
                {
            type: "password",
            name: "newPassword",
            label: "Create New Password",
            hasMatch: true,
            matchLabel: "Confirm New Password",
            validation: {
                required: true,
                validateFnName: "password",
                validationMsg: "Please enter a valid password.",
                requiredMsg: "Please enter a password.",
                nonMatchingMsg: "Passwords must match. Please try again.",
                requirementsLabel: "Your password must include",
                requirements: [
                    {name: "shortPassword", msg: "at least 8 characters"},
                    {name: "noUppercase", msg: "at least 1 uppercase letter"},
                    {name: "noLowercase", msg: "at least 1 lowercase letter"},
                    {name: "noDigits", msg: "at least 1 number"},
                    {name: "noSpecial", msg: "at least 1 symbol (for example, !, $, #, %)"}
                ]
            }
        }
    ]
}

export const troubleSigningInConfig = {
	submitEndpoint: "https://test-www.waters.com:8443/api/waters/user/v1/reset/password?email={email}",
	changePasswordEndpoint: "https://test-www.waters.com:8443/api/waters/user/v1/change/password",
	buttonText: "RESET PASSWORD",
	redirectUrl: "/content/waters/language-masters/en/reg-check-email.html",
	icons: {
		checkmarkIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
		validIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
		invalidIcon: "/content/dam/waters/en/brand-assets/icons/attention.svg",
		signInIcon: "/content/dam/waters/en/brand-assets/icons/user.svg"
	},
	fields: [
		{
			type: "email",
			name: "email",
			label: "Email Address}",
			validation: {
				required: true,
				requiredMsg: "Please enter a valid email address.",
				validateFnName: "email",
				validationMsg: "Please enter a valid email address."
			}
		},
		{
			type: "captcha",
			name: "captcha",
			siteKey: "6Ld5WMIUAAAAACZQvEc7I75aEg5AC8YUUO0W7zRG",
			validation: {
				required: true
			}
		}
	]
}

export const personalConfig = {
    submitEndpoint: "https://test-www.waters.com:8443/api/waters/user/v1/update/profile",
    buttonText: "Save Changes",
    cancelText: "Cancel",
    icons: {
        checkmarkIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        validIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        invalidIcon: "/content/dam/waters/en/brand-assets/icons/attention.svg",
        lockIcon: "/content/dam/waters/en/brand-assets/icons/lock.svg"
    },
    fields: [
        {
            type: "text",
            name: "firstName",
            label: "First Name",
            validation: {
                required: true,
                validateFnName: "noWhitespaceOnly",
                validationMsg: "Please enter a valid first name.",
                requiredMsg: "Please enter a first name."
            }
        },
        {
            type: "text",
            name: "lastName",
            label: "Last Name",
            validation: {
                required: true,
                validateFnName: "noWhitespaceOnly",
                validationMsg: "Please enter a valid last name.",
                requiredMsg: "Please enter a last name."
            }
        },
        {
            type: "text",
            name: "company",
            label: "Company",
            validation: {
                validateFnName: "noWhitespaceOnly",
                validationMsg: "Please enter a valid company name.",
                required: true,
                requiredMsg: "Please enter a company name."
            }
        },
        {
            type: "email",
            name: "email",
            disabled: true,
            label: "Email Address",
            validation: {
                required: true,
                requiredMsg: "Please enter a valid email address."
            }
        },
        {
            type: "text",
            name: "phone",
            label: "Phone Number",
            validation: {
               required: false
            }
        },
        {

            type: "dropdown",
            name: "country",
            label: "Country",
            placeholder: "Select...",
            dropdownIndicator: "/content/dam/waters/en/brand-assets/icons/down.svg",
            options: [{countryCode:"gb",displayName:"United Kingdom"},{countryCode:"us",displayName:"United States"}],
            "disabled": true,
            validation: {
                required: true,
                requiredMsg: "Please select your country."
            }
        },
        {
            "type": "checkbox",
            "name": "communications",
            label: "I want to receive communications from Waters.",
            validation: {
               required: true
            }
        }
    ]
}

export const currentPasswordConfig =  {
    type: "password",
    name: "currentPassword",
    label: "Enter Current Password",
    hasMatch: false,
    validation: {
        required: true,
        requiredMsg: "Please enter current password."
    }
}

export const signInConfig =
{
    "submitEndpoint": "https://test-www.waters.com:8443/api/waters/user/v1/login",
    "siteKey": "6Ld5WMIUAAAAACZQvEc7I75aEg5AC8YUUO0W7zRG",
    "formName": "signin",
    "buttonText": "SIGN IN",
    "icons": {
        "checkmarkIcon": "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        "validIcon": "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        "invalidIcon": "/content/dam/waters/en/brand-assets/icons/attention.svg",
        "eyeIcon": "/content/dam/waters/en/brand-assets/icons/eye.svg",
        "eyeOffIcon": "/content/dam/waters/en/brand-assets/icons/eye-off.svg",
        "signInIcon": "/content/dam/waters/en/brand-assets/icons/user.svg"
    },
    "fields": [
        {
            "type": "email",
            "name": "email",
            "label": "Email Address",
            "validation": {
                "required": true,
                "validateFnName": "email",
                "requiredMsg": "Please enter a valid email address.",
                "validationMsg": "Please enter a valid email address."
            }
        },
        {
            "type": "password",
            "name": "password",
            "label": "Password",
            "validation": {
                "required": true,
                "validateFnName": "noValidation",
                "requiredMsg": ""
            }
        },
        {
            "type": "link",
            "name": "forgotPassword",
            "text": "Having trouble signing in?",
            "link": "/nextgen/be.html"
        }
    ]
};


export const resetPasswordConfig = {
    submitEndpoint: "${resetPassword.changePasswordUrl @ context='unsafe'}",
    buttonText: "Reset Password",
    redirectUrl: "${resetPassword.redirectLink.href @ context='unsafe'}",
    icons: {
        checkmarkIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        validIcon: "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
        invalidIcon: "/content/dam/waters/en/brand-assets/icons/attention.svg",
        eyeIcon: "/content/dam/waters/en/brand-assets/icons/eye.svg",
        eyeOffIcon: "/content/dam/waters/en/brand-assets/icons/eye-off.svg",
        signInIcon: "/content/dam/waters/en/brand-assets/icons/user.svg"
    },
    fields: [
        {
            type: "body",
            name: "details",
            text: "You are resetting the password for the account associated with the email address",
            additionalText: {
                query: "email"
            }
        },
        {
            type: "password",
            name: "password",
            label: "Create Password",
            hasMatch: true,
            matchLabel: "Confirm Password",
            validation: {
                required: true,
                validateFnName: "password",
                validationMsg: "Please enter a valid password.",
                requiredMsg: "Please enter a password.",
                nonMatchingMsg: "Passwords must match. Please try again.",
                requirementsLabel: "Your password must include",
                requirements: [
                    {name: "shortPassword", msg: "at least 8 characters"},
                    {name: "noUppercase", msg: "at least 1 uppercase letter"},
                    {name: "noLowercase", msg: "at least 1 lowercase letter"},
                    {name: "noDigits", msg: "at least 1 number"},
                    {name: "noSpecial", msg: "at least 1 symbol (for example, !, $, #, %)"}
                ]
            }
        }
    ]
};
