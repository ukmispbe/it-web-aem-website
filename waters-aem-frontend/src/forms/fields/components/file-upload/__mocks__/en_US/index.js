export default {
    className: '',
    name: 'fileUploadInput',
    setRef: () => { },
    clearError: () => { },
    icons: {
        upload: '/content/dam/waters/en/brand-assets/icons/upload.svg',
        preview: '/content/dam/waters/en/brand-assets/icons/document.svg',
        valid: '/content/dam/waters/en/brand-assets/icons/success.svg',
        invalid: '/content/dam/waters/en/brand-assets/icons/x.svg'
    },
    chooseAFileToUpload: 'Choose a File to Upload',
    removeTextLabel: 'Remove',
    attachmentFileValidMsg: 'File format verified',
    attachmentFileInvalidValidMsg: 'File format not accepted',
    attachmentFileSize: '3MB',
    attachmentFileSizeErrorMsg: 'File size is over 5MB. Please try again.',
    maxAttachmentFileNameSizeWithExt: 32,
    attachmentFileNameLengthErrorMsg: 'File name is too long. Please try again.',
    attachmentFileNameErrorMsg: 'File name contains special characters. Please try again.',
    accept: '.pdf, .jpg, .png',
    fileTypePattern: '(\\.pdf|\.jpg|\.png)$'
};
