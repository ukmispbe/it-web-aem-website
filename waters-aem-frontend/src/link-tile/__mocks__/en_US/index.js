export default {
    title: "Test Link Tile",
    icon: "../../../public/attention.svg",
    links: [
        {
            "text": "Link One",
            "url": "http://www.example.com/link-one"
        },
        {
            "text": "Link Two",
            "url": "http://www.example.com/link-two"
        },
        {
            "text": "Link Three",
            "url": "http://www.example.com/link-three"
        }
    ]
};