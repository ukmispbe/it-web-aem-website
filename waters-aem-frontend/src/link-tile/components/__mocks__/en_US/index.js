export default {
    text: "Example Link",
    url: "http://www.example.com/"
};