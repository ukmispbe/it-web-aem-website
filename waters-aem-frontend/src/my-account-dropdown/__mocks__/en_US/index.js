import { MyOrderClass } from '../../my-account-item-list';

const DefaultProps = {
    config: {
        icon: '/content/dam/waters/en/brand-assets/icons/user.svg',
        closeIcon: '/content/dam/waters/en/brand-assets/icons/close.svg',
        title: 'My Account',
        notRegistered: 'Not registered yet?',
        signIn: {
            text: 'Sign In',
            url: '/content/waters/language-masters/en/signIn.html'
        },
        signOut: {
            text: 'Sign Out',
            url: '/content/waters/language-masters/en/signout.html'
        },
        switchAccount: {
            text: 'Switch Account',
            url: '/content/waters/language-masters/en/switch.html'
        },
        createAccount: {
            text: 'CREATE ACCOUNT',
            url: '/content/waters/language-masters/en/test.html'
        },
        itemList: [
            {
                text: 'My Profile',
                url: '/content/waters/language-masters/en/search.html',
                target: '_self',
                class: 'dropdown__item-list__my-profile'
            },
            {
                text: 'My Orders',
                url: '/content/waters/language-masters/en/library.html',
                target: '_self',
                class: MyOrderClass
            }
        ],
        loginState: true,
        userDetails: {
            userName: 'Susan Corman',
            accountName: 'Pfizer Randolf,',
            accountNumber: 'NJ 28264019282'
        }
    }
};

export default DefaultProps;
