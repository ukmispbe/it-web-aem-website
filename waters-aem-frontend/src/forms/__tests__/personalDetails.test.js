import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Form from '../form';

import { personalConfig } from '../__mocks__/en_US/mockData';
import { checkRenderInput, 
        checkRenderCheckbox, 
        checkRenderDropdown, 
        checkEventsInput, 
        checkEventsCheckbox } from '../__utils__/utils';

const mockSubmitFn = jest.fn();
const isocode = 'en_us';

let wrapper;
beforeEach(async () => {
    await act(async () => {
        wrapper = mount(<Form config={personalConfig} submitFn={mockSubmitFn} isocode={isocode} />);
    })
});

afterEach(() => {
    wrapper.unmount();
    jest.clearAllMocks();
});

describe('Feature: Personal Details Form', () => {
    describe('Scenario: Rendering', () => {
        describe('When initial render', () => {
            it('Then the snapshot should match', async () => {
                let json
                await renderer.act(async () => {
                    json = renderer.create(<Form config={personalConfig} submitFn={mockSubmitFn} isocode={isocode} />);
                });
                expect(json).toMatchSnapshot();
            });

            it('Then it should render a firstName field',  async ()=>{                 
                    checkRenderInput(wrapper, "firstName");
            });

            it('Then it should render a lastName field', async () => {
                checkRenderInput(wrapper, "lastName");
            });

            it('Then it should render a company field', async () => {
                checkRenderInput(wrapper, "company");
            });

            it('Then it should render an email field', async () => {
                checkRenderInput(wrapper, "email");
            });

            it('Then it should render a phone field', async () => {
                checkRenderInput(wrapper, "phone");
            });

            let isDisabled  = true;
            it('Then it should render a country field', async () => {
                checkRenderDropdown(wrapper, "country", isDisabled);
            });
            
            it('Then it should render a receive communications field', async () => {
                checkRenderCheckbox(wrapper, "communications");
            });

            it('Then it should render a disabled submit button', async () => {
                const button = wrapper.find('button');
                expect(button.exists()).toEqual(true);
                expect(button.instance().disabled).toEqual(true);
                expect(button.instance().type).toEqual("submit");
            });

            it('Then it should render a form', async () => {
                const form = wrapper.find('form');
                expect(form.exists()).toEqual(true);
            });

        });

        describe('Checking Events', () => {
            it('Then it should check events on firstName field',  async ()=> {
                checkEventsInput(wrapper, "firstName");                       
            });

            it('Then it should check events on lastName field', async () => {
                checkEventsInput(wrapper, "lastName");
            });

            it('Then it should check events on company field', async () => {
                checkEventsInput(wrapper, "company");
            });

            it('Then it should check events on email field', async () => {
                checkEventsInput(wrapper, "email");
            });

            it('Then it should check events on phone field', async () => {
                checkEventsInput(wrapper, "phone");
            });      
            
            it('Then it should check events on communications field', async () => {
                checkEventsCheckbox(wrapper, "communications");
            });
            
            it('Then it should check events on submit button', async () => {
                const button = wrapper.find('button');
                button.props().onClick = mockSubmitFn;               
                button.props().onClick();
                expect(mockSubmitFn).toHaveBeenCalledTimes(1);
            });

            it('Then it should check events on form', async () => {
                const form = wrapper.find('form');
                form.props().onSubmit = mockSubmitFn;               
                form.props().onSubmit();
                expect(mockSubmitFn).toHaveBeenCalledTimes(1);
            });

        });
    });
});