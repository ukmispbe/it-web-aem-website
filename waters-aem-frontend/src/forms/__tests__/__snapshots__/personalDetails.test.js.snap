// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Personal Details Form Scenario: Rendering When initial render Then the snapshot should match 1`] = `
<form
  className="cmp-form cmp-form--registration"
  onSubmit={[Function]}
>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      htmlFor="firstName"
    >
      First Name
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        className=""
        name="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      htmlFor="lastName"
    >
      Last Name
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        className=""
        name="lastName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      htmlFor="company"
    >
      Company
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        className=""
        name="company"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-email"
  >
    <label
      className=""
      htmlFor="email"
    >
      Email Address
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        className=""
        disabled={true}
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="email"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="lock-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      htmlFor="phone"
    >
      Phone Number
      <span
        className="cmp-form-field--optional"
      >
         
        (optional)
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        className=""
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-dropdown"
  >
    <div
      className="cmp-form-field-dropdown--wrapper"
    >
      <label
        htmlFor="country"
      >
        Country
      </label>
      <div
        className="cmp-form-field-dropdown--wrapper"
      >
        <div
          className="css-ssi0ig-container cmp-custom-dropdown--is-disabled"
          onKeyDown={[Function]}
        >
          <div
            className="css-1oonwhq-control cmp-custom-dropdown__control cmp-custom-dropdown__control--is-disabled"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="css-1hwfws3 cmp-custom-dropdown__value-container"
            >
              <div
                className="css-151xaom-placeholder cmp-custom-dropdown__placeholder"
              >
                Select...
              </div>
              <div
                className="css-1g6gooi"
              >
                <div
                  className="cmp-custom-dropdown__input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={true}
                    id="react-select-3-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className="css-1wy0on6 cmp-custom-dropdown__indicators"
            >
              <span
                className="css-1hyfx7x cmp-custom-dropdown__indicator-separator"
              />
              <div
                aria-hidden="true"
                className="css-16pqwjk-indicatorContainer cmp-custom-dropdown__indicator cmp-custom-dropdown__dropdown-indicator"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div />
              </div>
            </div>
          </div>
        </div>
        <div
          className="cmp-form-field--icons"
        >
          <div
            className="lock-icon"
          />
        </div>
      </div>
      <span
        className="cmp-form-field--errorText"
      >
        
      </span>
    </div>
  </div>
  <div
    className="cmp-form-field cmp-form-field-checkbox"
  >
    <input
      className="valid"
      id="communications"
      name="communications"
      readOnly={true}
      type="checkbox"
    />
    <a
      className="checkbox  valid"
      id="communications_link"
      onClick={[Function]}
    >
      <div />
    </a>
    <div
      className="cmp-form-field-checkbox--wrapper"
    >
      <label
        htmlFor="communications"
        onClick={[Function]}
      >
        I want to receive communications from Waters. 
      </label>
      
    </div>
  </div>
  <button
    className="cmp-button cmp-button--no-border cmp-form--submit cmp-button--disabled"
    disabled={true}
    type="submit"
  >
    Save Changes
  </button>
  <a
    className="cmp-button cmp-button--cancel"
    onClick={[Function]}
  >
    Cancel
  </a>
</form>
`;
